grails-app/domain/com/unifina/domain/marketplace/Product.groovy

```diff
@@ -28,6 +29,85 @@ class Product {
 	Integer score = 0 // set manually; used as default ordering for lists of Products (descending)
 	SecUser owner // set to product creator when product is created.
 
+	static class Contact {
+		// Contact's email address.
+		String email
+		// Contact's URL address.
+		String url
+		// Social media link 1
+		String social1
+		// Social media link 2
+		String social2
+		// Social media link 3
+		String social3
+		// Social media link 4
+		String social4
+
+		public Contact() {}
+		public Contact(Product product) {}
+
+		static constraints = {
+			email(nullable: true, validator: EmailValidator.validateNullEmail, maxSize: 255)
+			url(nullable: true, url: true, maxSize: 2048)
+			social1(nullable: true, url: true, maxSize: 2048)
+			social2(nullable: true, url: true, maxSize: 2048)
+			social3(nullable: true, url: true, maxSize: 2048)
+			social4(nullable: true, url: true, maxSize: 2048)
+		}
+
+		Map toMap() {
+			return [
+				email: email,
+				url: url,
+				social1: social1,
+				social2: social2,
+				social3: social3,
+				social4: social4,
+			]
+		}
+	}
+
+	static class TermsOfUse {
+		Boolean redistribution = true
+		Boolean commercialUse = true
+		Boolean reselling = true
+		Boolean storage = true
+		String termsUrl
+		String termsName
+
+		public TermsOfUse() {}
+		public TermsOfUse(Product product) {}
+
+		static constraints = {
+			redistribution(nullable: false)
+			commercialUse(nullable: false)
+			reselling(nullable: false)
+			storage(nullable: false)
+			termsUrl(nullable: true, url: true, maxSize: 2048)
+			termsName(nullable: true, maxSize: 100)
+		}
+		Map toMap() {
+			return [
+				redistribution: redistribution,
+				commercialUse: commercialUse,
+				reselling: reselling,
+				storage: storage,
+				termsUrl: termsUrl,
+				termsName: termsName,
+			]
+		}
+	}
+
+	// Product's contact details.
+	Contact contact
```

Comment:
Should these have default values? Would that avoid having to pass them in the `Spec`s?

```
Contact contact = new Contact()
TermsOfUse termsOfUse = new TermsOfUse()
```